gtp2ogs.js

```diff
@@ -1619,20 +1673,21 @@ class Connection {
             conn_log("Unhandled rules: " + notification.rules + ", rejecting challenge");
             return { reject: true, msg: "The " + notification.rules + " rules are not allowed for this bot, please choose allowed rules such as chinese rules. " };
         }
-        
-        
-        function timespanToDisplayString(timespan) {
-            let ss = timespan % 60;
-            let mm = Math.floor(timespan / 60 % 60);
-            let hh = Math.floor(timespan / (60*60) % 24);
-            let dd = Math.floor(timespan / (60*60*24));
-            let text = ["days", "hours", "minutes", "seconds"];
-            return [dd, hh, mm, ss]
-                .map((e, i) => e === 0 ? "" : `${e} ${text[i]}`)
-                .filter(e => e !== "")
-                .join(" ");
+
+        // ** first we eliminate absolute and simple time control : they don't support the use of a period time
+        // using absolute or simple (no period time) is likely to make bots timeout or play way too fast 
+        /* "fischer" , "simple", "byoyomi" , "canadian" , "absolute"*/
+        let TimeControlString = String(t.time_control);
+        if (TimeControlString === "absolute") {
+            conn_log("Minimum and Maximum main time is not supported in time control " + TimeControlString);
+            return { reject: true, msg: "Period time management is not supported in the time control " + TimeControlString + " , please choose a time control that supports the use of a period time, such as byoyomi,fischer,canadian." };
         }
     
+        if ((TimeControlString === "simple") || (TimeControlString === "simple")) {
```

Comment:
Why are we doing the same check twice here?

Reply:
mistake, was meant to be absolute+simple